src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -0,0 +1,65 @@
+using CUDA: has_cuda
+using OrderedCollections: OrderedDict
+
+using Oceananigans: AbstractModel, AbstractOutputWriter, AbstractDiagnostic
+
+using Oceananigans.Architectures: AbstractArchitecture
+using Oceananigans.Advection: CenteredSecondOrder
+using Oceananigans.BoundaryConditions: regularize_field_boundary_conditions
+using Oceananigans.Fields: Field, tracernames, VelocityFields, TracerFields
+using Oceananigans.Grids: with_halo
+using Oceananigans.TimeSteppers: Clock, TimeStepper
+using Oceananigans.Utils: inflate_halo_size, tupleit
+
+
+mutable struct ShallowWaterModel{G, A<:AbstractArchitecture, T, V, R, U, D, TS} <: AbstractModel{TS}
+    
+                 grid :: G         # Grid of physical points on which `Model` is solved
+         architecture :: A         # Computer `Architecture` on which `Model` is run
+                clock :: Clock{T}  # Tracks iteration number and simulation time of `Model`
+            advection :: V         # Advection scheme for velocities _and_ tracers
+             coriolis :: R         # Set of parameters for the background rotation rate of `Model`
+           velocities :: U         # Container for velocity fields `u`, and `v`
+              tracers :: D         # Container for tracer fields
+          timestepper :: TS        # Object containing timestepper fields and parameters
+end
+
+function ShallowWaterModel(;
+                           grid,
+  architecture::AbstractArchitecture = CPU(),
+                          float_type = Float64,
+                               clock = Clock{float_type}(0, 0, 1),
+                           advection = CenteredSecondOrder(),
+                            coriolis = nothing,
+                          velocities = nothing,
+                             tracers = (:D),
```

Comment:
The default `tracers` isn't a tuple, which may not be what you want. You may want to write

```julia
tracers = (:D,),
```

Note the comma, which ensures that the value is a tuple. You could also write

```julia
tracers = tuple(:D),
```

What does `:D` stand for?

Reply:
Before I realized I need to change set! I was thinking of D as depth, but I will change that and keep tracers as tracers.